docs/development/design/telemetry.rst

```diff
@@ -0,0 +1,232 @@
+Collect Data About Builds
+=========================
+
+We may want to take some decisions in the future about deprecations and supported versions.
+Right now we don't have data about the usage of packages and their versions on Read the Docs
+to be able to make a good decision.
+
+.. contents::
+   :local:
+   :depth: 3
+
+Data to be collected
+--------------------
+
+The following data can be collected after installing all dependencies.
+
+Configuration file
+~~~~~~~~~~~~~~~~~~
+
+We are saving the config file in our database,
+but to save some space we are saving it only if it's different than the one from a previous build
+(if it's the same we save a reference to it).
+
+The config file being saved isn't the original one used by the user,
+but the result of merging it with its default values.
+It's saved using a _fake_ ``JSONField``
+(charfield that is transformed to json when creating the model object).
+For these reasons we can't query or download them in bulk without iterating over all objects.
```

Comment:
I see how this is a problem, but it's not clear to me what the solution would be?

Reply:
Related: perhaps only certain keys of the configuration are of interest and this simplifies the problem? But I have no idea how are we parsing the `conf.py` at the moment.

Reply:
I think we need to upgrade Django to get real JSON fields, which let us query them.

Reply:
>  but it's not clear to me what the solution would be?

Use real json fields or another form to store this information